Linkta-org/core
# Implement functionality for users to Regenerate a new LinktaFlow

## Description
- Renamed the useNewUserInputMutation hook to useCreateLinktaFlowMutation for better clarity
- Updated the hook to utilize the pre-configured axiosClient
- Added functionality for the regenerate button in the UserInputList component
- Integrated the useCreateLinktaFlowMutation hook to handle the regeneration process
- Mapped userinput _id to id
- Implemented SnackBar Notification

Fixes #121 

## Type of change

## How Can this be tested? Testing Plan to review this PR
UserInputList
1. At root dir, run `npm run dev`
2. Selected one of the tabs on sidenav and click `Regenerate`
![tamagotchi](https://github.com/Linkta-org/core/assets/112212841/7ba13224-073d-45e4-9584-5948309b90ff)
4. A new tab with the same title will appear on the top of the navbar stack
![one piece](https://github.com/Linkta-org/core/assets/112212841/ebd11def-190a-4447-a1ba-8e5d15559398)
5. You will be routed to a new output path with the new userInputId

6. When encounter error, you will see an error toast notification:
![image](https://github.com/Linkta-org/core/assets/112212841/6a8bc36b-5308-4484-86b5-4f211d2b29bc)

**Note:** If you see a 500 it means the llm response has an error, try a few times until you get the 200

## Checklist:
